airsenal/framework/utils.py

```diff
@@ -238,19 +238,27 @@ def get_sell_price_for_player(player_id, gameweek=None, fpl_team_id=None):
     if the price increased in that time, we only get half the profit.
     if gameweek is None, get price we could sell the player for now.
     """
-    if not fpl_team_id:
-        fpl_team_id = fetcher.FPL_TEAM_ID
+    if fpl_team_id:
+        fetcher = FPLDataFetcher(fpl_team_id) # create new fetcher specific to team id
+    else:
+        fpl_team_id = fetcher.FPL_TEAM_ID #use saved fetcher with default team id.
+    
     transactions = session.query(Transaction)
     transactions = transactions.filter_by(fpl_team_id=fpl_team_id)
     transactions = transactions.filter_by(player_id=player_id)
     transactions = transactions.order_by(Transaction.gameweek).all()
 
+    history =fetcher.get_fpl_team_history_data()
+    starting_gw_adj = history['current'][0]['event'] - 1 #if you make a squad after then first gameweek then in transactions 'gameweek 1' is your starting gameweek
+
     gw_bought = None
     for t in transactions:
-        if gameweek and t.gameweek > gameweek:
+        if t.fpl_team_id != fpl_team_id: raise Exception(f'Transactions team id is {t.fpl_team_id} but cached data is for team id {fpl_team_id}')
+        trans_gameweek = t.gameweek + starting_gw_adj
```

Comment:
I think this is assuming that transaction gameweeks in the database are always relative to 1 (i.e. weeks since  team was created, rather than actual gameweeks)? I'm not sure that will be the case, the initial squad might be incorrectly listed as being transferred in gameweek 1, but the rest of the transactions should be assigned the correct gameweeks I believe.